### PR TITLE
Add Travis

### PR DESCRIPTION
**Pull request by mmha**
This requires `CONAN_USERNAME` to be set in the Travis config

### CODE REVIEW

File: config/conan/remotes.txt [hunk 1]
```diff
@@ -1,2 +1 @@
 conan-center https://api.bintray.com/conan/conan/conan-center True
-bincrafters https://api.bintray.com/conan/bincrafters/public-conan True
```
Comment (AerialMantis) [inline config/conan/remotes.txt:2]: Is this remote not required anymore?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+   os: linux
+   sudo: required
+   language: python
+   python: "3.6"
+   services:
+     - docker
+osx: &osx
+   os: osx
+   language: generic
+matrix:
```
Comment (AerialMantis) [inline .travis.yml:17]: Can we also add Windows to the test matrix?

**Review by AerialMantis — APPROVED** [review]
Looks good, thanks @mmha

File: .travis.yml (hunk 2 above)
Comment (mmha) [reply .travis.yml:17]: I can add AppVeyor in the same PR

File: config/conan/remotes.txt (hunk 1 above)
Comment (mmha) [reply config/conan/remotes.txt:2]: No, conan-center has everything we need now.

**Pull request closed without merging**